### ISSUE TITLE
Merge Direction and Direction 360

### ISSUE THREAD

**Issue by odysseywestra** [mypaint/mypaint#841]
We probably should merge the Direction and Direction 360 parameters of the brush engine. The only real difference is that one is limited 180 degrees and the other is full 360 degrees. This will help keep the brush engine gui consistent and less confusing.

@briend could you look into this since you helped bring in the Direction 360 into libmypaint?

**Comment by briend** [comment]
You know, it might actually "drop in" as a replacement without messing with any brushes.  I opened the Class/Knife brush and switched dab angle from Direction to Direction 360.  All I had to do was adjust the upper clipping on the 360 input to be 180 max, and then they seemed to be equivalent:
![image](https://user-images.githubusercontent.com/6015639/28257519-46c7d354-6a80-11e7-8cc6-780f51ef9f89.png)

If I'm not mistaken, the upper bound for the mapping is just the largest input in the brush file, so this would just happen on its own.   I'll round up a bunch of direction brushes and try this out to see what happens.

We should also make an internal alias so that direction_360 brushes can continue using that input, though.

**Comment by achadwick** [comment]
If it's a simple algorithmic mapping and it's going to be an actual replacement, you should write a convertor in the MyPaint code (lib.mypaint) before removing an input so that existing brush library behaviour will be retained.

**Comment by odysseywestra** [comment]
@briend @jplloyd This should be one of the issue that should be address before the 2.0 release. Mainly cause it introduces settings duplication.

**Comment by jplloyd** [comment]
I have brought this up elsewhere, but just to reiterate:

> The only real difference is that one is limited [to] 180 degrees and the other is full 360 degrees.

This is a bit of a misunderstanding of how these input types work and differ from each other. The original _Direction_ input will have the same value for strokes on the same line, regardless of direction (in other words, the same value for an angle `N` and `N +/- 180`).

_Direction 360_ naturally does not have this property, and would require the input mapping to be repeated (compressed and duplicated across the input range) in order emulate the behavior of _Direction_. One problem here is that since the number of input points are fixed, _Direction 360_ could not exactly replicate any _Direction_ mapping in the general case, but in practice it's probably not a real issue.

The real issue, in my opinion, is that removing _Direction_ in favor of _Direction 360_ would make it tedious to create brushes where you _want_ the behavior of _Direction_. You _could_ create a special interface for that mapping where you can switch types and have the graph replication done without the user having to know, but I don't really see what we gain from that compared to just having both input types available.

In short, I suggest we close this, but I'm always willing to be convinced that I'm wrong.

**Comment by odysseywestra** [comment]
With that in mind, it's probably best to go the direction of differentiating them through documentation and ui design. Right now any artist who just wants to tweak their brushes won't really understand the difference between the two so having  a distinction there would probably be a great help. 

I'd be willing to reword this issue if that the direction you two would rather go. I'm all for having all the settings, but each setting should be understandable by the average user.

**Comment by jplloyd** [comment]
Agreed, and even though I don't _think_ the average user edits brushes often, if at all, that doesn't make clarity any less important. 

Trivially the inputs are already differentiated by having different names. Changing the interface string for the original direction to `Direction 180` is one way to do it. For this change the translations could also be easily updated for most (but maybe not all) languages.

The tooltips are maybe a bit too condensed/terse given that we probably want them to clearly describe the differences between the two.

They currently are:

**Direction**: _The angle of the stroke, in degrees. The value will stay between 0.0 and 180.0, effectively ignoring turns of 180 degrees._

**Direction 360**: _The angle of the stroke, from 0 to 360 degrees._

As for external documentation, we need a section dedicated to brush editing in the wiki. As far as I can tell, brush editing has not been documented since 0.9, and even then not fully.

What did you have in mind for ui design?
